Mathlib/CategoryTheory/Category/Pairwise.lean

```diff
@@ -74,20 +74,13 @@ def comp : ∀ {o₁ o₂ o₃ : Pairwise ι} (_ : Hom o₁ o₂) (_ : Hom o₂
   | _, _, _, left i j, id_single _ => left i j
   | _, _, _, right i j, id_single _ => right i j
 
-section
-
-open Lean Elab Tactic in
-/-- A helper tactic for `aesop_cat` and `Pairwise`. -/
-def pairwiseCases : TacticM Unit := do
-  evalTactic (← `(tactic| casesm* (_ : Pairwise _) ⟶ (_ : Pairwise _)))
-
-attribute [local aesop safe tactic (rule_sets := [CategoryTheory])] pairwiseCases in
 instance : Category (Pairwise ι) where
   Hom := Hom
   id := id
-  comp f g := comp f g
-
-end
+  comp := @comp _
+  id_comp f := by cases f <;> rfl
+  comp_id f := by cases f <;> rfl
+  assoc f g _ := by cases f <;> cases g <;> rfl
```

Comment:
Do we know what happened here?

Reply:
Splitting the construction into `CategoryStruct` stage, and then a `Category` stage allows us to keep the automation:

```
instance : CategoryStruct (Pairwise ι) where
  Hom := Hom
  id := id
  comp := @comp _

section

open Lean Elab Tactic in
/-- A helper tactic for `aesop_cat` and `Pairwise`. -/
def pairwiseCases : TacticM Unit := do
  evalTactic (← `(tactic| casesm* (_ : Pairwise _) ⟶ (_ : Pairwise _)))

attribute [local aesop safe tactic (rule_sets := [CategoryTheory])] pairwiseCases in
instance : Category (Pairwise ι) where

end
```

Reply:
I've pushed this.

Reply:
My thinking here was that the automation was many more bytes than doing it directly, and the automation helped work around the fact that the fields weren't seeing the actual provided definition for `comp`. Maybe what I did was too big of a change for a bump PR though.